modules/dataframe/include/inviwo/dataframe/datastructures/column.h

```diff
@@ -64,12 +66,39 @@ class IVW_MODULE_DATAFRAME_API Column : public MetaDataOwner {
     virtual ~Column() = default;
 
     virtual Column* clone() const = 0;
+    virtual Column* clone(const std::vector<std::uint32_t>& rowSelection) const = 0;
 
     virtual ColumnType getColumnType() const = 0;
 
     virtual const std::string& getHeader() const = 0;
     virtual void setHeader(std::string_view header) = 0;
 
+    virtual Unit getUnit() const = 0;
+    virtual void setUnit(Unit unit) = 0;
+
+    /**
+     * Set a custom range for the column which can be used for normalization, plotting, color
+     * mapping, etc.
+     */
+    virtual void setCustomRange(dvec2 range) = 0;
+    virtual void clearCustomRange() = 0;
+
+    /**
+     * Returns the custom column range. If no range has been set previously, the return
+     * value will be std::nullopt.
+     */
+    virtual std::optional<dvec2> getCustomRange() const = 0;
+
+    /**
+     * Returns the range of the data, ignoring any custom range
+     */
+    virtual dvec2 getDataRange() const = 0;
+
+    /**
+     * Returns the custom range if set or else the range of the data
```

Comment:
The range of the data = min/max of the data? Might want to state that instead in that case